### ISSUE TITLE
Question: Custom Field: Prevent form from submit and show errors in error list on top of the form

### ISSUE THREAD

**Issue by svankireddy-cartera** [mozilla-services/react-jsonschema-form#1325]
This is a question not an issue.
I have create new custom field but when Error occurs in that custom field:
I dont want form to be submitted, Its not very clear in documentation on how to do it?

Use Case:
When Error Occurs on the custom field how Can we stop form from  submit and show error (I dont want to use validate instead write code for errors in same custom field )?

Note: this is not a runnable code its just for visual purpose
```


class CustomLinkFields extends Component {
  constructor(props, context) {
    super(props, context);
    console.log(props);
    this.state = {
      warning: '',
      showValidLinkMessage: false,
      validLink: false,
      value: props.value,
      contextRef: createRef(),
      required: props.required
    };
  }

  
  checkLink(event) {
   //When an error is detected I would Like to show an error and let the form know error has occured and restrict form to be submitted
  }

  handleChange(event) {
    this.setState({
      value: event.target.value
    }, () => this.props.onChange(this.state.value));
  }

  render() {
    let validLinkMessage = '';
    const {contextRef, value, warning, required, showValidLinkMessage, validLink} = this.state;
    const { schema: {title} , idSchema: {$id}} = this.props;

    return (
      <div className="custom-link-widget" ref={contextRef}>
        <label className="control-label" htmlFor={$id}>{title}{required && <span className="required">*</span>}</label>
        <input type="string"
               className="form-control"
               value={value}
               required={required}
               onChange={this.handleChange.bind(this)}
               onBlur={(event) => this.checkLink(event)}/>
        {warning && <FadeOutMessage message={warning} contextRef={contextRef}/>}
        {validLinkMessage}
      </div>
    );
  }
}

export default CustomLinkFields;
```

Any help is greatly appreciated :)

**Comment by epicfaace** [comment]
@svankireddy-cartera what do you mean by "prevent form from submit"? You have full control of what happens during form submit through the `onSubmit` handler.

Could you link to a reproducible example on jsfiddle so it becomes clearer what exactly you are trying to do?

**Comment by epicfaace** [comment]
You _can_ use the `showErrorList` prop to control where the error list is shown, though: https://react-jsonschema-form.readthedocs.io/en/latest/validation/#error-list-display

**Comment by phatNfqAsia** [comment]
@epicfaace how about the other question `When Error Occurs on the custom field, how can we stop form from submit and show error?` Have this lib covered this use cases? I am interested in this. Like when we have to show a disabled SUBMIT button
